src/main/java/com/icthh/xm/tmf/ms/communication/service/TelegramService.java

```diff
@@ -0,0 +1,131 @@
+package com.icthh.xm.tmf.ms.communication.service;
+
+import static com.icthh.xm.tmf.ms.communication.utils.LogUtil.withLog;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.icthh.xm.commons.logging.aop.IgnoreLogginAspect;
+import com.icthh.xm.tmf.ms.communication.channel.telegram.TelegramUpdateListener;
+import com.icthh.xm.tmf.ms.communication.config.ApplicationProperties;
+import com.icthh.xm.tmf.ms.communication.domain.CommunicationSpec.Telegram;
+import com.icthh.xm.tmf.ms.communication.domain.MessageType;
+import com.icthh.xm.tmf.ms.communication.web.api.model.CommunicationMessage;
+import com.icthh.xm.tmf.ms.communication.web.api.model.CommunicationMessageCreate;
+import com.icthh.xm.tmf.ms.communication.web.api.model.Receiver;
+import com.pengrad.telegrambot.TelegramBot;
+import com.pengrad.telegrambot.request.SendMessage;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.time.StopWatch;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@Service
+@IgnoreLogginAspect
+@RequiredArgsConstructor
+public class TelegramService implements MessageService {
+
+    private final ApplicationProperties applicationProperties;
+    private final KafkaTemplate<String, String> kafkaTemplate;
+    private final ObjectMapper objectMapper;
+
+    private Map<String, Map<String, TelegramBot>> tenantTelegramBots = new ConcurrentHashMap<>();
+
+    public void registerBot(String tenantKey, Telegram botConfig) {
+        Map<String, TelegramBot> telegramBots = getTenantBots(tenantKey);
+        if (telegramBots.containsKey(botConfig.getKey())) {
+            log.info("[{}] Skip telegram bot registration because such bot already registered: [{}]",
```

Comment:
suggest to set level to WARN

Reply:
this method will be called on each configuration update, so I think it's normal log level in this case